Code/GraphMol/StructChecker/StructChecker.h

```diff
@@ -0,0 +1,266 @@
+  void setAcidicAugmentedAtoms(const std::vector<AugmentedAtom> &acidicAtoms);
+
+  bool loadGoodAugmentedAtoms(const std::string &path);
+  void setGoodAugmentedAtoms(const std::vector<AugmentedAtom> &acidicAtoms);
+
+  bool loadPatterns(const std::string &path);  // file with clean patterns
+  void parsePatterns(
+      const std::vector<std::string> &smarts);  // can throw RDKit exeptions
+  void setPatterns(const std::vector<ROMOL_SPTR> &p);
+
+  bool loadRotatePatterns(
+      const std::string &path);  // file with rotate patterns
+  void parseRotatePatterns(
+      const std::vector<std::string> &smarts);  // can throw RDKit exeptions
+  void setRotatePatterns(const std::vector<ROMOL_SPTR> &p);
+
+  bool loadStereoPatterns(
+      const std::string &path);  // file with stereo patterns
+  void parseStereoPatterns(
+      const std::vector<std::string> &smarts);  // can throw RDKit exeptions
+  void setStereoPatterns(const std::vector<ROMOL_SPTR> &p);
+
+  bool loadTautomerData(const std::string &path);  // file path
+  void parseTautomerData(const std::vector<std::string> &smartsFrom,
+                         const std::vector<std::string> &smartsTo);
+  void setTautomerData(const std::vector<ROMOL_SPTR> &from,
+                       const std::vector<ROMOL_SPTR> &to);
+  bool loadChargeDataTables(const std::string &path);  // file path
+};
+
+bool parseOptionsJSON(const std::string &json, StructCheckerOptions &op);
+
+bool loadOptionsFromFiles(
+    StructCheckerOptions &op,
+    const std::string &augmentedAtomTranslationsFile = "",
+    // ?? AcidicAtoms;
+    // ?? GoodAtoms;
+    const std::string &patternFile = "",        // file with clean patterns
+    const std::string &rotatePatternFile = "",  // file with rotate patterns
+    const std::string &stereoPatternFile = "",  // file with stereo patterns
+    const std::string &tautomerFile = "");
+
+class StructChecker {
+ public:
+  typedef enum StructureFlags {
+    NO_CHANGE = 0,
+    BAD_MOLECULE = 0x0001,
+    ALIAS_CONVERSION_FAILED = 0x0002,
+    STEREO_ERROR = 0x0004,
+    STEREO_FORCED_BAD = 0x0008,
+    ATOM_CLASH = 0x0010,
+    ATOM_CHECK_FAILED = 0x0020,
+    SIZE_CHECK_FAILED = 0x0040,
+    // reserved error = 0x0080,
+    TRANSFORMED = 0x0100,
+    FRAGMENTS_FOUND = 0x0200,
+    EITHER_WARNING = 0x0400,
+    DUBIOUS_STEREO_REMOVED = 0x0800,
+    RECHARGED = 0x1000,
+    STEREO_TRANSFORMED = 0x2000,
+    TEMPLATE_TRANSFORMED = 0x4000,
+    TAUTOMER_TRANSFORMED = 0x8000,
+    // mask:
+    BAD_SET = (BAD_MOLECULE | ALIAS_CONVERSION_FAILED | STEREO_ERROR |
+               STEREO_FORCED_BAD | ATOM_CLASH | ATOM_CHECK_FAILED |
+               SIZE_CHECK_FAILED),
+
+    TRANSFORMED_SET = (TRANSFORMED | FRAGMENTS_FOUND | EITHER_WARNING |
+                       DUBIOUS_STEREO_REMOVED | STEREO_TRANSFORMED |
+                       TEMPLATE_TRANSFORMED | TAUTOMER_TRANSFORMED | RECHARGED),
+  } StructureFlags;
+  // attributes:
+ private:
+  const StructCheckerOptions Options;
```

Comment:
Does this need to be const?
It would be nice to be able to change parameters for things like checking stereo or verbosity in an pre-constructed `StructChecker` object.
If this doesn't need to be const, please remove that specification and add a method that returns a reference to the Options structure so that parameters can be updated.